hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip904/TokenClaimAirdropTest.java

```diff
@@ -117,6 +140,98 @@ final Stream<DynamicTest> claimFungibleTokenAirdrop() {
                         getAccountInfo(RECEIVER).hasToken(relationshipWith(NON_FUNGIBLE_TOKEN)));
     }
 
+    @HapiTest
+    @DisplayName("token claim with no pending airdrop should fail")
+    final Stream<DynamicTest> tokenClaimWithNoPendingAirdrop() {
+        return hapiTest(tokenClaimAirdrop().hasPrecheck(EMPTY_PENDING_AIRDROP_ID_LIST));
+    }
+
+    @HapiTest
+    @DisplayName("token claim with duplicate entires should fail")
+    final Stream<DynamicTest> duplicateClaimAirdrop() {
+        return hapiTest(
+                cryptoCreate(OWNER).balance(ONE_HUNDRED_HBARS).maxAutomaticTokenAssociations(0),
+                cryptoCreate(OWNER_2).balance(ONE_HUNDRED_HBARS).maxAutomaticTokenAssociations(0),
+                cryptoCreate(RECEIVER).balance(ONE_HUNDRED_HBARS),
+                tokenCreate(FUNGIBLE_TOKEN)
+                        .treasury(OWNER)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                tokenCreate(FUNGIBLE_TOKEN_2)
+                        .treasury(OWNER_2)
+                        .tokenType(FUNGIBLE_COMMON)
+                        .initialSupply(1000L),
+                newKeyNamed(NFT_SUPPLY_KEY),
+                tokenCreate(NON_FUNGIBLE_TOKEN)
+                        .name(NON_FUNGIBLE_TOKEN)
+                        .treasury(OWNER)
+                        .tokenType(NON_FUNGIBLE_UNIQUE)
+                        .initialSupply(0L)
+                        .supplyKey(NFT_SUPPLY_KEY),
+                mintToken(NON_FUNGIBLE_TOKEN, List.of(ByteString.copyFromUtf8("a"), ByteString.copyFromUtf8("b"))),
+                tokenAirdrop(moving(10, FUNGIBLE_TOKEN).between(OWNER, RECEIVER))
+                        .payingWith(OWNER),
+                tokenAirdrop(moving(20, FUNGIBLE_TOKEN_2).between(OWNER_2, RECEIVER))
+                        .payingWith(OWNER_2),
+                tokenAirdrop(movingUnique(NON_FUNGIBLE_TOKEN, 1).between(OWNER, RECEIVER))
+                        .payingWith(OWNER),
+                tokenClaimAirdrop(
+                                pendingAirdrop(OWNER, RECEIVER, FUNGIBLE_TOKEN),
+                                pendingAirdrop(OWNER, RECEIVER, FUNGIBLE_TOKEN),
+                                pendingAirdrop(OWNER_2, RECEIVER, FUNGIBLE_TOKEN_2),
+                                pendingNFTAirdrop(OWNER, RECEIVER, NON_FUNGIBLE_TOKEN, 1))
+                        .payingWith(RECEIVER)
+                        .feeUsd(0.001)
+                        .via("claimTxn")
+                        .hasPrecheck(PENDING_NFT_AIRDROP_ALREADY_EXISTS));
+    }
```

Comment:
There is no need to send airdrops in this case.
Here, when you send claim transaction, it should fail in the `pureChecks` with PENDING_AIRDROP_ID_REPEATED, so no state checks.

This way the test will be way simpler and easy to read
```suggestion
    @HapiTest
    @DisplayName("token claim with duplicate entires should fail")
    final Stream<DynamicTest> duplicateClaimAirdrop() {
        return hapiTest(
                cryptoCreate(RECEIVER).balance(ONE_HUNDRED_HBARS),
                tokenClaimAirdrop(
                                pendingAirdrop(OWNER, RECEIVER, FUNGIBLE_TOKEN),
                                pendingAirdrop(OWNER, RECEIVER, FUNGIBLE_TOKEN))
                        .payingWith(RECEIVER)
                        .feeUsd(0.001)
                        .hasPrecheck(PENDING_AIRDROP_ID_REPEATED));
    }
```